primaryHandler.go

```diff
@@ -158,7 +157,22 @@ func authChain(v *viper.Viper, logger log.Logger, registry xmetrics.Registry) (a
 		basculehttp.WithEErrorResponseFunc(listener.OnErrorResponse),
 	)
 
-	constructors := []alice.Constructor{setLogger(logger), authConstructor, authEnforcer, basculehttp.NewListenerDecorator(listener)}
+	constructors := []alice.Constructor{setLogger(logger,
+
+		// custom logger func that extracts the intended destination of requests
+		func(kv []interface{}, request *http.Request) []interface{} {
```

Comment:
Does this function require any variables defined in its parent function? I can't see any so I would recommend naming/declaring it outside of this function and then passing it here.